transport_nantes/photo/static/photo/js/vote_system.js

```diff
@@ -0,0 +1,120 @@
+function getCookie(name) {
+    let cookieValue = null;
+    if (document.cookie && document.cookie !== '') {
+        const cookies = document.cookie.split(';');
+        for (let i = 0; i < cookies.length; i++) {
+            const cookie = cookies[i].trim();
+            // Does this cookie string begin with the name we want?
+            if (cookie.substring(0, name.length + 1) === (name + '=')) {
+                cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+                break;
+            }
+        }
+    }
+    return cookieValue;
+}
+
+
+/**
+ * Cast vote when the user has already voted on any photo
+ * @param {boolean} vote_nature "upvote" or "downvote"
+ * @param {HTMLElement} element "the button that has been pressed to cast vote"
+ */
+function castVote(vote_nature, element) {
+    $.ajax({
+        url: window.location.href,
+        headers: {"X-CSRFToken": getCookie("csrftoken")},
+        type: 'post',
+        data: {
+            vote_nature: vote_nature,
+        },
+        mode: 'same-origin',
+        success: () => {
+            console.log("Success");
+            applySelectedVoteStyles(element);
+        },
+        error: (data) => {
+            console.log("Error");
+            console.log(data);
+        }
+    });
+}
+/**
+ * This function is called on the first vote for a user (anon or not)
```

Comment:
Why do we need a different function for the first vote than for later votes?  This deserves explanation.

Reply:
066960a18bfe86819aa60577be33c2de17f1cf5b

This was moved in base template in this commit, are you sure you are looking at the correct head ? (6ccba43c7b90726c75f11feb36294877bf1e99ed)

Reply:
We don't need the same information the first time and the next times. 
The first time we have vote_nature and at least consent box, and if not authenticated, also email and captcha.  These are retrieved with a modal form that appears when you click on any vote button when you never voted. 

The second form, once you voted a first time, doesn't require a modal and doesn't send the same info, just the vote_nature. Doing so you can vote without filling a modal and it's quicker.